src/components/Maps/Sidebar.tsx

```diff
@@ -62,14 +72,25 @@ const Sidebar = ({
   const styles = useStyles();
 
   useEffect(() => {
+    if (clickedAddress) {
+      setAddress(clickedAddress);
+    }
+
+    if (clickedLocation) {
+      setLatitude(clickedLocation.lat);
+      setLongitude(clickedLocation.lng);
+    }
+
     if (editLabel) {
       setLabel(editLabel);
     }
 
-    if (editAddress) {
+    if (editAddress && editLocation && address === '') {
       setAddress(editAddress);
+      setLatitude(editLocation.lat);
+      setLongitude(editLocation.lng);
     }
-  }, [editLabel, editAddress]);
+  }, [clickedAddress, clickedLocation, editLabel, editAddress]);
```

Comment:
![image](https://user-images.githubusercontent.com/59516223/102727684-f4f68600-42db-11eb-8aee-8aa66723c633.png)
obligatory please double check this works instead of making any assumptions
```suggestion
  }, [clickedAddress, clickedLocation, editLabel, editAddress, editLocation, address]);
```